src/main/kotlin/com/pubnub/api/subscribe/eventengine/effect/EmitMessagesProvider.kt

```diff
@@ -0,0 +1,25 @@
+package com.pubnub.api.subscribe.eventengine.effect
+
+import com.pubnub.api.models.consumer.PNStatus
+import com.pubnub.api.models.consumer.pubsub.PNMessageResult
+import com.pubnub.api.models.consumer.pubsub.PNPresenceEventResult
+import com.pubnub.api.models.consumer.pubsub.PNSignalResult
+import com.pubnub.api.models.consumer.pubsub.files.PNFileEventResult
+import com.pubnub.api.models.consumer.pubsub.message_actions.PNMessageActionResult
+import com.pubnub.api.models.consumer.pubsub.objects.PNObjectEventResult
+
+interface EmitMessagesProvider {
```

Comment:
The name `Provider` is usually associated with something that returns something. None of the methods in it return anything so I don't think it should be named `Provider` at all

Reply:
ok, so maybe EmitMessagesConsumer ?